### PR TITLE
created separate perms for windows so that windows users will no longer experience permission denied issues

### PR DESCRIPTION
**Pull request by isemaya-square**
Context: https://square.slack.com/archives/C03C3457C01/p1652248777115799?thread_ts=1650924230.904889&cid=C03C3457C01

TL;DR windows does not allow 440 permissions (which were being set for key files in certstap). In reality, the key files were being given 444 permissions.

However, this was previously allowed because file permissions checking in certstrap was flawed - it was allowing files with looser permissions than expected to be used. This was fixed in https://github.com/square/certstrap/pull/141. However, the side effect of this fix was that certstrap broke in windows due to 444 (the actual key file permissions) being looser than 440. This PR fixes this by setting special leaf perms to be 444 for windows only.

### CODE REVIEW

**Review by jdtw — APPROVED** [review]
Test this on Windows if you can?
